contracts/exchangeIssuance/ExchangeIssuanceLeveraged.sol

```diff
@@ -0,0 +1,1159 @@
+     * @return Amount of output token returned to the user
+     */
+    function _liquidateCollateralTokens(
+        uint256 _collateralTokenSpent,
+        ISetToken _setToken,
+        uint256 _setAmount,
+        address _originalSender,
+        Exchange _exchange,
+        address _outputToken,
+        uint256 _minAmountOutputToken,
+        address _collateralToken,
+        uint256 _collateralAmount,
+        SwapData memory _swapData
+    )
+    internal
+    returns(uint256)
+    {
+        require(_collateralAmount >= _collateralTokenSpent, "ExchangeIssuance: OVERSPENT COLLATERAL TOKEN");
+        uint256 amountToReturn = _collateralAmount.sub(_collateralTokenSpent);
+        uint256 outputAmount;
+        if(_outputToken == ETH_ADDRESS){
+            outputAmount = _liquidateCollateralTokensForETH(
+                _collateralToken,
+                amountToReturn,
+                _exchange,
+                _originalSender,
+                _minAmountOutputToken,
+                _swapData
+            );
+        } else {
+            outputAmount = _liquidateCollateralTokensForERC20(
+                _collateralToken,
+                amountToReturn,
+                _exchange,
+                _originalSender,
+                IERC20(_outputToken),
+                _minAmountOutputToken,
+                _swapData
+            );
+        }
+        emit ExchangeRedeem(_originalSender, _setToken, _outputToken, _setAmount, outputAmount);
+        return outputAmount;
+    }
+
+    /**
+     * Returns the collateralToken directly to the user
+     *
+     * @param _collateralToken       Address of the the collateral token
+     * @param _amountToReturn        Amount of the underlying collateral token to return
+     * @param _originalSender        Address of the original sender to return the tokens to
+     * @param _minAmountOutputToken  Minimum amount of output token to return to the user
+     */
+    function _returnCollateralTokensToSender(
+        address _collateralToken,
+        uint256 _amountToReturn,
+        address _originalSender,
+        uint256 _minAmountOutputToken
+    )
+    internal
+    {
+            IERC20(_collateralToken).transfer(_originalSender, _amountToReturn);
+    }
+
+    /**
+     * Sells the collateral tokens for the selected output ERC20 and returns that to the user
+     *
+     * @param _collateralToken       Address of the collateral token
+     * @param _amountToReturn        Amount of the underlying collateral token to return
+     * @param _exchange              Enum indicating which exchange to use
+     * @param _originalSender        Address of the original sender to return the tokens to
+     * @param _outputToken           Address of token to return to the user
+     * @param _minAmountOutputToken  Minimum amount of output token to return to the user
+     */
+    function _liquidateCollateralTokensForERC20(
+        address _collateralToken,
+        uint256 _amountToReturn,
```

Comment:
I could call it `excessCollateralToSell` but that is not 100% accurate since, if the user chooses the collateral token as output no sale is performed.
I can call it `excessCollateralToSellOrReturn` but that seems a bit wordy. 
What do you think ?

Reply:
Changed name to `_collateralRemaining` to indicate that this is the collateral remaining after swapping a share of the original amount for the debt token.